### DISCUSSION TITLE
fatal: could not read Username, Device not configured

### DISCUSSION THREAD

**Discussion by dsonzini** [Vinzent03/obsidian-git#316] [Problems] [answered]
Hi, this is happening to me when I try to git push on obsidian. Tried to look for solutions but couldn't find anything.

fatal: could not read Username for 'https://github.com': Device not configured

Thanks

**Comment by Vinzent03** [comment, marked as answer]
This comment lists all your options for authentication https://github.com/denolehov/obsidian-git/issues/254#issuecomment-1173180978
